pytorch_lightning/trainer/trainer.py

```diff
@@ -911,8 +911,9 @@ def __test_using_best_weights(self, ckpt_path, test_dataloaders):
                     f'specify a path for a checkpoint .test(ckpt_path=PATH)'
                 )
                 return {}
-            if not self._device_type == DeviceType.TPU:
-                self.accelerator.barrier()
+
+            if torch.distributed.is_available() and torch.distributed.is_initialized():
+                self.training_type_plugin.barrier()
```

Comment:
would it make sense to move the check for `torch.distributed` into the specific barrier method in the plugin?
Then only the plugins that use distributed would run this check and others will do nothing

Reply:
Good idea !